Unable to build with GCC

tested and failing on `g++11` ([gh actions Ubuntu](https://github.com/dumbasPL/linux-pe-repo/actions/runs/12508970708/job/34897824277)) and `g++14` (Arch Linux)

Here is a minimal example: https://github.com/dumbasPL/linux-pe-repo

Seems like `sizeof( symbol_t )` is 20 on GCC for some reason. Works with clang but I would expect a project with `linux` in its name to work with the most common linux compiler.

LOG:
```
[1/2] Building CXX object CMakeFiles/linux-pe-repo.dir/main.cpp.o
FAILED: CMakeFiles/linux-pe-repo.dir/main.cpp.o 
/usr/bin/g++  -I/home/nezu/git/linux-pe-repo/linux-pe/includes -g -std=gnu++20 -MD -MT CMakeFiles/linux-pe-repo.dir/main.cpp.o -MF CMakeFiles/linux-pe-repo.dir/main.cpp.o.d -o CMakeFiles/linux-pe-repo.dir/main.cpp.o -c /home/nezu/git/linux-pe-repo/main.cpp
In file included from /home/nezu/git/linux-pe-repo/linux-pe/includes/coff/image.hpp:36,
                 from /home/nezu/git/linux-pe-repo/linux-pe/includes/linuxpe:2,
                 from /home/nezu/git/linux-pe-repo/main.cpp:1:
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/section_header.hpp:99:91: warning: multi-character character constant [-Wmultichar]
   99 |             return !characteristics.mem_not_paged && ( *( uint32_t* ) &name.short_name == 'ade.' || *( uint32_t* ) &name.short_name == 'EGAP' );
      |                                                                                           ^~~~~~
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/section_header.hpp:99:136: warning: multi-character character constant [-Wmultichar]
   99 |             return !characteristics.mem_not_paged && ( *( uint32_t* ) &name.short_name == 'ade.' || *( uint32_t* ) &name.short_name == 'EGAP' );
      |                                                                                                                                        ^~~~~~
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/section_header.hpp:103:90: warning: multi-character character constant [-Wmultichar]
  103 |             return characteristics.mem_discardable || *( uint32_t* ) &name.short_name == 'TINI';
      |                                                                                          ^~~~~~
In file included from /home/nezu/git/linux-pe-repo/linux-pe/includes/coff/image.hpp:33:
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/symbol.hpp:138:39: error: static assertion failed: Invalid enum bitfield.
  138 |     static_assert( sizeof( symbol_t ) == 18, "Invalid enum bitfield." );
      |                    ~~~~~~~~~~~~~~~~~~~^~~~~
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/symbol.hpp:138:39: note: the comparison reduces to ‘(20 == 18)’
In file included from /home/nezu/git/linux-pe-repo/linux-pe/includes/coff/image.hpp:37:
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/auxiliaries/aux_file_name.hpp:45:46: error: static assertion failed: Invalid auxiliary symbol.
   45 |     static_assert( sizeof( aux_file_name_t ) == sizeof( symbol_t ), "Invalid auxiliary symbol." );
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
/home/nezu/git/linux-pe-repo/linux-pe/includes/coff/auxiliaries/aux_file_name.hpp:45:46: note: the comparison reduces to ‘(18 == 20)’
In file included from /home/nezu/git/linux-pe-repo/linux-pe/includes/nt/image.hpp:32,
                 from /home/nezu/git/linux-pe-repo/linux-pe/includes/linuxpe:3:
/home/nezu/git/linux-pe-repo/linux-pe/includes/nt/directories/dir_exceptions.hpp:146:45: warning: ‘win::unwind_info_t::frame_register’ is too small to hold all values of ‘enum class win::unwind_register_id’
  146 |                 unwind_register_id          frame_register : 4;
      |                                             ^~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

--- a/includes/img_common.hpp
+++ b/includes/img_common.hpp
@@ -45,7 +45,7 @@
     #endif
 #endif
 
-#pragma pack(push, 1)
+COFF_STRUCT_PACKING
 namespace win
 {
     // Default image architecture

